src/dealDashboard/dealDashboard.ts

```diff
@@ -29,15 +29,10 @@ export class DealDashboard {
       this.ethereumService.defaultAccountAddress &&
       (
         /* Open deal can not be private, and anyone can comment and begin a discussion */
-        !this.deal.registrationData.partnerDAO
+        this.deal.isOpenProposal ||
         /* In a Partnered deal only members can comment and begin a discussion */
-        || [
-          this.deal.registrationData.proposalLead?.address,
-          ...this.deal.registrationData.primaryDAO?.representatives.map(representative => representative.address) || "",
-          ...this.deal.registrationData.partnerDAO?.representatives.map(representative => representative.address) || "",
-        ].includes(this.ethereumService.defaultAccountAddress)
+        [...this.deal.representativesAndLead].includes(this.ethereumService.defaultAccountAddress)
```

Comment:
First, note the correct say to use Set.  But you don't need this anyway, use `deal.isRepresentativeUser` (which does it corrrectly)